File: 前端/vue-A13/src/types/job.ts
// 岗位相关类型定义

export interface Job {
  id: string
  title: string
  company: string
  salary: string
  location: string
  industry: string
  experience: string
  education: string
  description: string
  tags: string[]
  createdAt: string
}

export interface JobProfile {
  jobId: string
  professionalSkills: string[]
  certificates: string[]
  softSkills: string[]
  developmentPath: string[]
  responsibilities: string[]
  requirements: string[]
}

export interface JobGraphNode {
  id: string
  name: string
  type: string
  level: number
}

export interface JobGraphLink {
  source: string
  target: string
  relationship: string
}

export interface JobGraph {
  nodes: JobGraphNode[]
  links: JobGraphLink[]
}

export interface JobSearchParams {
  keyword?: string
  industry?: string
  salaryMin?: number
  salaryMax?: number
  experience?: string
  education?: string
  sortBy?: 'salary' | 'popularity' | 'date'
  page?: number
  pageSize?: number
}

File: 后端/models/job.py
# app/models/job.py
from sqlalchemy import Column, Integer, String, Text, Float
from database import Base

class Job(Base):
    __tablename__ = "jobs"

    id = Column(Integer, primary_key=True, index=True)
    title = Column(String(200), nullable=False)
    company = Column(String(200), nullable=False)
    salary = Column(String(100))
    location = Column(String(200))
    experience = Column(String(100))
    education = Column(String(100))
    description = Column(Text)
    requirements = Column(Text)
    tags = Column(String(500))
    industry = Column(String(200))
    type = Column(String(100))
    publish_date = Column(String(100))